lib/web/join_tokens.go

```diff
@@ -245,6 +291,45 @@ func getJoinScript(settings scriptSettings, m nodeAPIGetter) (string, error) {
 	return buf.String(), nil
 }
 
+// generateIamTokenName makes a deterministic name for a iam join token
+// based on its rule set
+func generateIamTokenName(rules []*types.TokenRule) (string, error) {
+	// sort the rules by (account ID, arn)
+	// to make sure a set of rules will produce the same hash,
+	// no matter the order they are in the slice
+	orderedRules := make([]*types.TokenRule, len(rules))
+	copy(orderedRules, rules)
+	sortRules(orderedRules)
+
+	var sb strings.Builder
+	for _, r := range orderedRules {
+		s := fmt.Sprintf("%s-%s", r.AWSAccount, r.AWSARN)
```

Comment:
I don't think the hyphen is necessary right? doesn't matter now but I'm imagining if we need to add more fields to the hash, and we want old hashes to remain the same, we won't be able to unconditionally add another hyphen.

You could probably skip the strings.Builder and write everything directly to the hash

Reply:
Yes, right, that makes sense.
Added on https://github.com/gravitational/teleport/pull/11339/commits/2fea8f1a6967cf534610de725c1b767bcfd7e737, let me know if it is what you had in mind